backend/Application/Api/GraphQL/ChainParametersV3.cs

```diff
@@ -0,0 +1,51 @@
+using Application.Api.GraphQL.Accounts;
+using Application.Api.GraphQL.Extensions;
+
+namespace Application.Api.GraphQL;
+
+public class ChainParametersV3 : ChainParameters, IEquatable<ChainParametersV3>
+{
+    public ulong RewardPeriodLength { get; init; }
+
+    internal static ChainParametersV3 From(Concordium.Sdk.Types.ChainParametersV3 input)
+    {
+        return new ChainParametersV3
+        {
+            EuroPerEnergy = ExchangeRate.From(input.EuroPerEnergy),
+            MicroCcdPerEuro = ExchangeRate.From(input.MicroCcdPerEuro),
+            AccountCreationLimit = (int)input.AccountCreationLimit.Limit,
+            FoundationAccountAddress = AccountAddress.From(input.FoundationAccount),
+            RewardPeriodLength = input.TimeParameters.RewardPeriodLength.RewardPeriodEpochs.Count,
+        };
+
+    }
+
+    public bool Equals(ChainParametersV3? other)
+    {
+        return other != null &&
+               base.Equals(other) &&
+               RewardPeriodLength == other.RewardPeriodLength;
+    }
+
+    public override bool Equals(object? obj)
+    {
+        if (ReferenceEquals(null, obj)) return false;
+        if (ReferenceEquals(this, obj)) return true;
+        return obj.GetType() == GetType() && Equals(obj as ChainParametersV2);
```

Comment:
You have to validate that obj is of type ChainParametersV2.

Reply:
I assume you basically just have to cast to V3 instead.

Reply:
Nice catch